packages/clay-dropdown/src/ClayDropdownBase.js

```diff
@@ -2,17 +2,153 @@ import 'clay-button';
 import 'clay-checkbox';
 import 'clay-icon';
 import 'clay-radio';
-import Dropdown from 'metal-dropdown';
-import Soy from 'metal-soy';
+import {Align} from 'metal-position';
+import {core, object} from 'metal';
 import {Config} from 'metal-state';
-
+import Component from 'metal-component';
+import dom from 'metal-dom';
+import {EventHandler} from 'metal-events';
+import Soy from 'metal-soy';
 import itemsValidator from './items_validator';
 import templates from './ClayDropdownBase.soy.js';
 
 /**
  * Implementation of the base for Metal Clay Dropdown.
  */
-class ClayDropdownBase extends Dropdown {
+class ClayDropdownBase extends Component {
+	/**
+	 * @inheritDoc
+	 */
+	attached() {
+		super.attached();
+		this.eventHandler_.add(
+			dom.on(document, 'click', this.handleDocClick_.bind(this))
+		);
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	created() {
+		this.eventHandler_ = new EventHandler();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	detached() {
+		super.attached();
+		this.eventHandler_.removeAllListeners();
+	}
+
+	/**
+	 * Closes the dropdown.
+	 * @protected
+	 */
+	close_() {
+		this.expanded = false;
+	}
+
+	/**
+	 * Handles document click in order to hide menu.
+	 * @param {!Event} event
+	 * @protected
+	 */
+	handleDocClick_(event) {
+		if (this.element.contains(event.target)) {
+			return;
+		}
+		this.close_();
+	}
+
+	/**
+	 * Toggles the dropdown, closing it when open or opening it when closed.
+	 */
+	toggle() {
+		this.expanded = !this.expanded;
+	}
+
+	/**
+	 * The setter function for the `classMap_` staet.
+	 * @param {Object} value
+	 * @return {!object}
+	 * @protected
+	 */
+	setterClassMapFn_(value) {
+		return object.mixin(this.valueClassMapFn_(), value);
+	}
+
+	/**
+	 * The setter function for the `position` state. Converts the supported
+	 * string positions into the appropriate `Align` position constants.
+	 * @param {string|number} value
+	 * @return {number}
+	 * @protected
+	 */
+	setterPositionFn_(value) {
+		if (core.isNumber(value)) {
+			return value;
+		}
+		return value.toLowerCase() === 'up' ? Align.TopLeft : Align.BottomLeft;
+	}
+
+	/**
+	 * Synchronization logic for `expanded` state.
+	 * @param {boolean} expanded
+	 */
+	syncExpanded(expanded) {
+		if (expanded && this.alignElementSelector) {
+			// eslint-disable-next-line
```

Comment:
We should try avoiding to disable line linting. In this case we can just split the line in two:

```
let alignElement = this.element.querySelector(
   this.alignElementSelector);
```

Reply:
@carloslancha I had done that, but the **prettier** is leaving everything inline, had to add to not break in the CI.

Reply:
We merged https://github.com/prettier/prettier-eslint/pull/141 yesterday and a new version of `prettier-eslint` was auto-released. It should solve the issue.

@matuzalemsteles, can you try updating the dependencies to see if this goes away?

Reply:
Hey @jbalsas, it is necessary that `prettier-eslint-cli` update the version of `prettier-eslint` to **8.2.2**.

Reply:
@Robert-Frampton, could you take care of that?

Reply:
@jbalsas for some reason the `npm run checkFormat` is failing.

Reply:
It's because some files are out of format. See the log:

```
packages/clay-alert/src/ClayAlert.js
packages/clay-charts/src/ChartBase.js
packages/clay-collapse/src/ClayCollapse.js
packages/clay-dropdown/src/items_validator.js
packages/clay-modal/src/ClayModal.js
packages/clay-collapse/src/__tests__/ClayCollapse.js
packages/clay-navbar/src/__tests__/ClayNavbar.js
success formatting 7 files with prettier-eslint
```

If you run `npm run format` locally, would you probably see those changes. The `checkFormat` task is there to ensure nothing gets committed with the wrong format.

Can you run that and see what's changing? You can run the `npm run checkFormat` task before sending this over to see if some files need formatting.

Reply:
This was due to a `yarn upgrade prettier-eslint-cli` which upgraded `prettier-eslint` to **8.2.2** and had to make some changes. So as not to mess up the `yarn.lock` tree, I think we'd better keep it out of this, until` prettier-eslint-cli` is using **8.2.2**. what do you think ?

Reply:
Sounds good!